hikari/impl/cache.py

```diff
@@ -114,16 +113,11 @@ class StatefulCacheImpl(cache.MutableCache):
         cache_utility.GenericRefWrapper[emojis.CustomEmoji],
     ]
     _user_entries: collections.ExtendedMutableMapping[snowflakes.Snowflake, cache_utility.GenericRefWrapper[users.User]]
-    _message_entries: collections.LimitedCapacityCacheMap[snowflakes.Snowflake, messages.Message]
-    _intents: intents_.Intents
-
-    def __init__(
-        self, app: traits.RESTAware, intents: intents_.Intents, settings: typing.Optional[config.CacheSettings]
-    ) -> None:
-        if settings is None:
-            settings = config.CacheSettings()
+    _message_entries: collections.ExtendedMutableMapping[snowflakes.Snowflake, messages.Message]
 
+    def __init__(self, app: traits.RESTAware, settings: config.CacheSettings) -> None:
         self._app = app
+        self._settings = settings
```

Comment:
might be nice to have some way to introspectively check the settings on the cache if we know which type it is. (so just impl detail where you'd have to instance check it first rather than as a part of the abc or protocol)